src/main/Main.hs

```diff
@@ -788,6 +789,20 @@ execCmd ExecOpts {..} go@GlobalOpts{..} =
           pkgopts <- getPkgOpts menv wc pkgs
           return (prefix ++ compilerExeName wc, pkgopts ++ args)
 
+      runWithPath path callback = case path of
+        Nothing -> callback
+        Just p | not (isValid p) -> callback
```

Comment:
Hmm, it seems like this would silently ignore `--cwd` for invalid paths.  Is there a reason for this?  Seems to me like it should throw an error.